### PR TITLE
Fix bug in ipaddr netmask filter when cidr is 32

### LINKED ISSUE #12065: ipaddr filter returns empty netmask for host address
##### ISSUE TYPE
Bug Report

##### COMPONENT NAME
core

##### ANSIBLE VERSION
```
ansible 1.9.1
```

##### CONFIGURATION
nothing

##### OS / ENVIRONMENT
Ubuntu 14.04.2 LTS

##### SUMMARY
ipaddr() filter incorrectly return empty string for ipv4 host netmask (/32). It should return "255.255.255.255".

##### STEPS TO REPRODUCE
hosts:
```
127.0.0.1 ip=1.1.1.1/32
```
site.yml
```
---
- hosts: all
  gather_facts: false
  tasks:
  - debug: var="{{ ip | ipaddr('netmask')  }}"
```
```
$ansible-playbook -i hosts site.yml
```

##### EXPECTED RESULTS
```
ok: [127.0.0.1] => {
    "var": {
        "255.255.255.255": "255.255.255.255"
    }
}
```

##### ACTUAL RESULTS
```
ok: [127.0.0.1] => {
    "var": {
        "": ""
    }
}
```

### PR DESCRIPTION
**Pull request by ryanpineo**
##### SUMMARY
Fixes #12065 

##### ISSUE TYPE
Feature Pull Request

##### COMPONENT NAME
lib/ansible/plugins/filter/ipaddr.py
test/units/plugins/filter/test_ipaddr.py
test/utils/tox/requirements-py3.txt
test/utils/tox/requirements.txt

##### ANSIBLE VERSION
2.3

### CODE REVIEW

**Comment by ryanpineo** [comment]
The tests are partially failing but due to no fault of my own.  Wondering what the best course of action here is.

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ addons:
     packages:
       - python2.4
 install:
-  - pip install tox PyYAML Jinja2 sphinx
+  - pip install tox PyYAML Jinja2 sphinx netaddr
```
Comment (bcoca) [inline .travis.yml]: many plugins have extra dependencies, we do not add those to the main ansible ones

Comment (egroeper) [inline .travis.yml]: Then how to write tests for i.e. core filter plugins?
Isn't netaddr needed to test this core filter plugin?
Should the tests be dropped?

Comment (sivel) [inline .travis.yml]: The test requirements are maintained in https://github.com/ansible/ansible/blob/devel/test/utils/tox/requirements.txt

Comment (egroeper) [inline .travis.yml]: Thanks for the hint!
So by simply moving the netaddr test dependency from .travis.yml to tox/requirements.txt everything should be fine?

Then I will try to make a new PR based on this.

Comment (ryanpineo) [inline .travis.yml]: I'll update this

**Comment by ryanpineo** [comment]
Updated PR to use the test requirements.txt file

**Comment by mattclay** [comment]
@ryanpineo Could you update this PR to use `test/runner/requirements/units.txt` for requirements? We no longer use `test/utils/tox/` for CI.

**Comment by abadger** [comment]
A different fix has been applied here: https://github.com/ansible/ansible/pull/17627 and is present in stable-2.2 and beyond.  The test cases in your PR are good, though.  I've rebased and updated to the units.txt file that mattclay mentioned and pushed a new PR with those: https://github.com/ansible/ansible/pull/23506

**Pull request closed without merging**